lib/beaker-puppet/install_utils/foss_utils.rb

```diff
@@ -826,16 +826,28 @@ def install_puppet_agent_from_dmg_on(hosts, opts)
         # Returns the latest puppet-agent version number from a given url.
         #
         # @param [String] url         URL containing list of puppet-agent packages.
-        #                             Example: https://downloads.puppetlabs.com/mac/10.11/PC1/x86_64
+        #                             Example: https://downloads.puppetlabs.com/mac/puppet7/10.15/x86_64/
         #
         # @return [String] version    puppet-agent version number (e.g. 1.4.1)
         #                             Empty string if none found.
         # @api private
         def get_latest_puppet_agent_build_from_url(url)
           require 'oga'
           require 'net/http'
-          body = Net::HTTP.get(URI.parse("#{url}/index_by_lastModified_reverse.html"))
-          document = Oga.parse_html(body)
+
+          full_url = "#{url}/index_by_lastModified_reverse.html"
+          response = Net::HTTP.get_response(URI(full_url))
+          counter = 0
+
+          # Redirect following
+          while response.code == '301' && counter < 15
```

Comment:
There are a few different redirection HTTP codes, so you could check for each one, something like `[301, 302, 307, ...].include?(response.code)` or check the response class, e.g. `if response.is_a?(Net::HTTPRedirection)`